lucene/core/src/java/org/apache/lucene/search/IndriQuery.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.search;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Basic abstract query that all IndriQueries can extend to implement
+ * toString, equals, getClauses, and iterator.
+ *
+ */
+public abstract class IndriQuery extends Query
```

Comment:
Random question: will `IndriQuery` take advantage of Block MAX Weak And optimization?  The added `smoothingScore` must alter the optimization logic to find the min block score to skip to?

I think it's OK to defer BMW to followon improvements, as long as it is not kicking in incorrectly here.

Reply:
Currently, IndriQuery does not take advantage of Block MAX Weak and optimization.  We iterate through all documents that that have a posting for at least one of the search terms.  I would be interested in expanding the smoothing score functionality to more parts of lucene in the future.